src/types.ts

```diff
@@ -1,71 +1,160 @@
+  step_codename?: string
+  step_num?: number
+  user_choice?: string
+  source?: string
+  form_name?: string
+  real_signup_error_message?: string
+  landing_company: string
 }
 
 type VirtualSignupEmailConfirmation = {
-    action?: 'received' | 'expired' | 'confirmed' | 'error'
-    signup_provider?: SignupProvider
-    form_source?: string
-    email_md5?: string
-    error_message?: string
+  action?: 'received' | 'expired' | 'confirmed' | 'error'
+  signup_provider?: SignupProvider
+  form_source?: string
+  email_md5?: string
+  error_message?: string
 }
 type TradeTypesForm = {
-    action?: 'open' | 'close' | 'choose_trade_type' | 'search' | 'info_open' | 'info-switcher' | 'info_close'
-    trade_type_name?: string
-    tab_name?: string
-    search_string?: string
-    info_switcher_mode?: 'description' | 'glossary'
-    form_source?: string
-    subform_name?: string
-    account_type?: string
+  action?:
+    | 'open'
+    | 'close'
+    | 'choose_trade_type'
+    | 'search'
+    | 'info_open'
+    | 'info_switcher'
+    | 'info_close'
+  trade_type_name?: string
+  tab_name?: string
+  search_string?: string
+  info_switcher_mode?: 'description' | 'glossary'
+  form_source?: string
+  subform_name?: string
+  account_type?: string
 }
 type DbotTypes = {
-    action?: 'open' | 'close' | 'search' | 'delete' | 'yes' | 'no'| 'search_string' | 'choose_shortcut' | 'bot_last_modified_time' | 'delete_popup_respond' | 'push_open_button' | 'push_user_guide' | 'save_your_bot' | 'choose_your_bot' | 'delete_your_bot'
-    shortcut_name?: string
-    form_source?: string
-    search_string?: string
-    delete_popup_respond?: string
-    bot_last_modified_time?: number
-    bot_name?: string
-};
+  action?:
+    | 'open'
+    | 'close'
+    | 'search'
+    | 'delete'
+    | 'yes'
+    | 'no'
+    | 'search_string'
+    | 'choose_shortcut'
+    | 'bot_last_modified_time'
+    | 'delete_popup_respond'
+    | 'push_open_button'
+    | 'push_user_guide'
+    | 'save_your_bot'
+    | 'choose_your_bot'
+    | 'delete_your_bot'
+  shortcut_name?: string
+  form_source?: string
+  search_string?: string
+  delete_popup_respond?: string
+  bot_last_modified_time?: number
+  bot_name?: string
+}
+
+type IndicatorsTypesFormAction = {
+  action:
+    | 'open'
+    | 'close'
+    | 'add_active'
+    | 'clean_all_active'
+    | 'delete_active'
+    | 'edit_active'
+    | 'search'
+    | 'info_open'
+    | 'info_close'
+  form_name?: string
+  indicator_type_name?: string
+  indicators_category_name?: string
+  search_string?: string
+  subform_name?: string
+  account_type: string
+  device_type: string
+}
+
+type MarketTypesFormAction = {
+  action:
+    | 'open'
+    | 'close'
+    | 'choose_market_type'
+    | 'search'
+    | 'info_redirect'
+    | 'add_to_favorites'
+    | 'delete_from_favorites'
+  form_name: string
+  market_type_name: string
+  search_string?: string
+  tab_market_name?: string
+  account_type: string
+  device_type: string
```

Comment:
this will be moved in extra 'core schema' and will provide in separate function because this data is repeatable everywhere we track events
if you can and want, please remove these params from other types 🙏

Reply:
let me check if we are even using this in our `trackEvent` calls

Reply:
269-270 lines
https://github.com/binary-com/deriv-app/pull/10183/files#diff-8e9c0ba264688ff6e53beb3f7972fe5a661715c527a9d94fd3f15cd8f4cd47b3